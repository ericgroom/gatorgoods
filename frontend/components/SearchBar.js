import styled from "styled-components";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";

const Form = styled.form`
  button {
    background-color: ${props => props.theme.yellow};
  }
`;

class SearchBar extends React.Component {
  state = {
    category:
      this.props.defaultCategory ||
      (this.props.categories && this.props.categories[0].value),
    query: this.props.defaultQuery || ""
  };
  handleChange = e => {
    console.log(e);
    this.setState({ [e.target.name]: e.target.value });
  };
  handleSubmit = e => {
    e.preventDefault();
    this.props.onSearch(this.state.category);
  };
  render() {
    return (
      <Form onSubmit={this.handleSubmit}>
        <div className="input-group">
          <div className="input-group-prepend">
            <select
              className="custom-select"
              id="category-select"
              name="category"
              value={this.state.category}
              onChange={this.handleChange}
            >
              {this.props.categories &&
                this.props.categories.map(category => (
                  <option
                    className="dropdown-item"
                    value={category.value}
                    key={category.value}
                  >
                    {category.name}
                  </option>
                ))}
            </select>
          </div>
          <input
            type="text"
            id="search-input"
            name="query"
            className="form-control"
            placeholder="Search for an item..."
            value={this.state.query}
            onChange={this.handleChange}
          />
          <div className="input-group-append">
            <button className="btn btn-outline" type="submit">
              <FontAwesomeIcon icon={faSearch} fixedWidth />
            </button>
          </div>
        </div>
      </Form>
    );
  }
}

SearchBar.propTypes = {
  categories: PropTypes.array,
  onSearch: PropTypes.func
};

export default SearchBar;
