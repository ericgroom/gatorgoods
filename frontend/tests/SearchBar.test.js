import SearchBar from "../components/SearchBar";
import { shallow, render, mount } from "enzyme";

const fakeCategories = [
  { name: "All", value: 0 },
  { name: "Books", value: 1 },
  { name: "Furniture", value: 2 }
];

describe("SearchBar component", () => {
  it("renders without props", () => {
    expect(() => render(<SearchBar />)).not.toThrow();
  });
  it("renders with categories", () => {
    const wrapper = shallow(<SearchBar categories={fakeCategories} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("can be passed a default category selection", () => {
    const wrapper = shallow(
      <SearchBar categories={fakeCategories} defaultCategory={2} />
    );
    expect(wrapper.state("category")).toBe(2);
  });
  it("will set the default category selection to the first element otherwise", () => {
    const wrapper = shallow(<SearchBar categories={fakeCategories} />);
    expect(wrapper.state("category")).toBe(fakeCategories[0].value);
  });
  it("calls prop-provided method onSearch", () => {
    const handleSearch = jest.fn();
    const wrapper = mount(
      <SearchBar categories={fakeCategories} onSearch={handleSearch} />
    );
    expect(wrapper.find("form").exists()).toBe(true);
    const form = wrapper.find("form").first();
    form.simulate("submit");
    expect(handleSearch.mock.calls[0][0]).toBe(0);
  });
  it("can be passed initial selections", () => {
    const wrapper = shallow(
      <SearchBar
        categories={fakeCategories}
        defaultCategory={1}
        defaultQuery="shoes"
      />
    );
    expect(wrapper.state("query")).toBe("shoes");
    expect(wrapper.state("category")).toBe(1);
  });
});
