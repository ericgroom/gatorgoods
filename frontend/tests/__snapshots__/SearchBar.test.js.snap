// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders with categories 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    categories={
      Array [
        Object {
          "name": "All",
          "value": 0,
        },
        Object {
          "name": "Books",
          "value": 1,
        },
        Object {
          "name": "Furniture",
          "value": 2,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="input-group"
      >
        <div
          className="input-group-prepend"
        >
          <select
            className="custom-select"
            id="category-select"
            name="category"
            onChange={[Function]}
            value={0}
          >
            <option
              className="dropdown-item"
              value={0}
            >
              All
            </option>
            <option
              className="dropdown-item"
              value={1}
            >
              Books
            </option>
            <option
              className="dropdown-item"
              value={2}
            >
              Furniture
            </option>
          </select>
        </div>
        <input
          className="form-control"
          id="search-input"
          name="query"
          onChange={[Function]}
          placeholder="Search for an item..."
          type="text"
          value=""
        />
        <div
          className="input-group-append"
        >
          <button
            className="btn btn-outline"
            type="submit"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={true}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </button>
        </div>
      </div>,
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="input-group-prepend"
          >
            <select
              className="custom-select"
              id="category-select"
              name="category"
              onChange={[Function]}
              value={0}
            >
              <option
                className="dropdown-item"
                value={0}
              >
                All
              </option>
              <option
                className="dropdown-item"
                value={1}
              >
                Books
              </option>
              <option
                className="dropdown-item"
                value={2}
              >
                Furniture
              </option>
            </select>
          </div>,
          <input
            className="form-control"
            id="search-input"
            name="query"
            onChange={[Function]}
            placeholder="Search for an item..."
            type="text"
            value=""
          />,
          <div
            className="input-group-append"
          >
            <button
              className="btn btn-outline"
              type="submit"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={true}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </button>
          </div>,
        ],
        "className": "input-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <select
              className="custom-select"
              id="category-select"
              name="category"
              onChange={[Function]}
              value={0}
            >
              <option
                className="dropdown-item"
                value={0}
              >
                All
              </option>
              <option
                className="dropdown-item"
                value={1}
              >
                Books
              </option>
              <option
                className="dropdown-item"
                value={2}
              >
                Furniture
              </option>
            </select>,
            "className": "input-group-prepend",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  className="dropdown-item"
                  value={0}
                >
                  All
                </option>,
                <option
                  className="dropdown-item"
                  value={1}
                >
                  Books
                </option>,
                <option
                  className="dropdown-item"
                  value={2}
                >
                  Furniture
                </option>,
              ],
              "className": "custom-select",
              "id": "category-select",
              "name": "category",
              "onChange": [Function],
              "value": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": "All",
                  "className": "dropdown-item",
                  "value": 0,
                },
                "ref": null,
                "rendered": "All",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": "Books",
                  "className": "dropdown-item",
                  "value": 1,
                },
                "ref": null,
                "rendered": "Books",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": "Furniture",
                  "className": "dropdown-item",
                  "value": 2,
                },
                "ref": null,
                "rendered": "Furniture",
                "type": "option",
              },
            ],
            "type": "select",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "id": "search-input",
            "name": "query",
            "onChange": [Function],
            "placeholder": "Search for an item...",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-outline"
              type="submit"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={true}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </button>,
            "className": "input-group-append",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={true}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />,
              "className": "btn btn-outline",
              "type": "submit",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": true,
                "flip": null,
                "icon": Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                },
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "SearchBar__Form-sc-1qxhise-0",
        "isStatic": false,
        "lastClassName": "cGpBhk",
        "rules": Array [
          "button{background-color:",
          [Function],
          ";}",
        ],
      },
      "displayName": "SearchBar__Form",
      "render": [Function],
      "styledComponentId": "SearchBar__Form-sc-1qxhise-0",
      "target": "form",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="input-group"
        >
          <div
            className="input-group-prepend"
          >
            <select
              className="custom-select"
              id="category-select"
              name="category"
              onChange={[Function]}
              value={0}
            >
              <option
                className="dropdown-item"
                value={0}
              >
                All
              </option>
              <option
                className="dropdown-item"
                value={1}
              >
                Books
              </option>
              <option
                className="dropdown-item"
                value={2}
              >
                Furniture
              </option>
            </select>
          </div>
          <input
            className="form-control"
            id="search-input"
            name="query"
            onChange={[Function]}
            placeholder="Search for an item..."
            type="text"
            value=""
          />
          <div
            className="input-group-append"
          >
            <button
              className="btn btn-outline"
              type="submit"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={true}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </button>
          </div>
        </div>,
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="input-group-prepend"
            >
              <select
                className="custom-select"
                id="category-select"
                name="category"
                onChange={[Function]}
                value={0}
              >
                <option
                  className="dropdown-item"
                  value={0}
                >
                  All
                </option>
                <option
                  className="dropdown-item"
                  value={1}
                >
                  Books
                </option>
                <option
                  className="dropdown-item"
                  value={2}
                >
                  Furniture
                </option>
              </select>
            </div>,
            <input
              className="form-control"
              id="search-input"
              name="query"
              onChange={[Function]}
              placeholder="Search for an item..."
              type="text"
              value=""
            />,
            <div
              className="input-group-append"
            >
              <button
                className="btn btn-outline"
                type="submit"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={true}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f002",
                        "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                      ],
                      "iconName": "search",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </button>
            </div>,
          ],
          "className": "input-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <select
                className="custom-select"
                id="category-select"
                name="category"
                onChange={[Function]}
                value={0}
              >
                <option
                  className="dropdown-item"
                  value={0}
                >
                  All
                </option>
                <option
                  className="dropdown-item"
                  value={1}
                >
                  Books
                </option>
                <option
                  className="dropdown-item"
                  value={2}
                >
                  Furniture
                </option>
              </select>,
              "className": "input-group-prepend",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    className="dropdown-item"
                    value={0}
                  >
                    All
                  </option>,
                  <option
                    className="dropdown-item"
                    value={1}
                  >
                    Books
                  </option>,
                  <option
                    className="dropdown-item"
                    value={2}
                  >
                    Furniture
                  </option>,
                ],
                "className": "custom-select",
                "id": "category-select",
                "name": "category",
                "onChange": [Function],
                "value": 0,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "All",
                    "className": "dropdown-item",
                    "value": 0,
                  },
                  "ref": null,
                  "rendered": "All",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Books",
                    "className": "dropdown-item",
                    "value": 1,
                  },
                  "ref": null,
                  "rendered": "Books",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Furniture",
                    "className": "dropdown-item",
                    "value": 2,
                  },
                  "ref": null,
                  "rendered": "Furniture",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "id": "search-input",
              "name": "query",
              "onChange": [Function],
              "placeholder": "Search for an item...",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-outline"
                type="submit"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={true}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f002",
                        "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                      ],
                      "iconName": "search",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </button>,
              "className": "input-group-append",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={true}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f002",
                        "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                      ],
                      "iconName": "search",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
                "className": "btn btn-outline",
                "type": "submit",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "",
                  "fixedWidth": true,
                  "flip": null,
                  "icon": Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                    ],
                    "iconName": "search",
                    "prefix": "fas",
                  },
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "SearchBar__Form-sc-1qxhise-0",
          "isStatic": false,
          "lastClassName": "cGpBhk",
          "rules": Array [
            "button{background-color:",
            [Function],
            ";}",
          ],
        },
        "displayName": "SearchBar__Form",
        "render": [Function],
        "styledComponentId": "SearchBar__Form-sc-1qxhise-0",
        "target": "form",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
